src/helm/proxy/clients/http_client.py

```diff
@@ -0,0 +1,151 @@
+from dataclasses import asdict
+
+from helm.common.cache import Cache, CacheConfig
+from helm.common.request import (
+    Request,
+    RequestResult,
+    Sequence,
+    Token,
+    EMBEDDING_UNAVAILABLE_REQUEST_RESULT,
+)
+from helm.common.tokenization_request import (
+    DecodeRequest,
+    DecodeRequestResult,
+    TokenizationRequest,
+    TokenizationRequestResult,
+    TokenizationToken,
+)
+from .client import Client, wrap_request_time
+
+import requests
+
+
+class HTTPClient(Client):
+    """Implements a simple HTTP client."""
+
+    def __init__(
+        self,
+        cache_config: CacheConfig,
+        base_url: str = "http://localhost",
+        port: int = 8080,
+        timeout: int = 10,
+        do_cache: bool = False,
+    ):
+        self.cache = Cache(cache_config)
+        self.base_url = base_url
+        self.port = port
+        self.timeout = timeout
+        self.do_cache = do_cache
+
+    def make_request(self, request: Request) -> RequestResult:
+        cache_key = asdict(request)
+        # This needs to match whatever we define in pedantic
+        if request.embedding:
+            return EMBEDDING_UNAVAILABLE_REQUEST_RESULT
+
+        # Only a single stop sequence is supported as we can only pass in a single value for `eos_token_id`
+        if len(request.stop_sequences) > 1:
+            raise ValueError("More than one stop sequence is not supported.")
+
+        raw_request = {
+            "prompt": request.prompt,
+            "temperature": 1e-7 if request.temperature == 0 else request.temperature,
```

Comment:
Do we need this with the model setup? Otherwise just set this to `request.temperature`. An alternative would be to handle this server-side instead.